tritium-processor/src/test/resources/com/palantir/tritium/examples/InstrumentedBindsParameter.java.generated

```diff
@@ -61,6 +61,10 @@ public final class InstrumentedBindsParameter implements BindsParameter {
                 BindsParameter.class, delegate, InstrumentedBindsParameter::new);
     }
 
+    public static BindsParameter trace(BindsParameter delegate) {
+        return InstrumentationBuilder.trace(BindsParameter.class, delegate, InstrumentedBindsParameter::new);
+    }
```

Comment:
I'd gone back and forth on adding something like this initially -- it probably doesn't make sense to write a tracing-java-specific processor, however that would avoid the InvocationContext overhead incurred by tritium (that said, it would likely be best in the long term to evolve the tritium handler api to avoid that requirement, and centralize this sort of thing in one project)

Reply:
> > Noticed in some JFR that @Instrument generated classes were grow() the
> > InstrumentationBuilder.handlers list when there were only 1 or 2 handlers being used per instance.
> 
> I wouldn't generally expect instrumentation wrappers to be created on a per-operation basis, usually they're created at application startup. The annotation processor certainly has much lower overhead than the reflective proxy or byte-buddy implementations, but I want to make sure we're not doing something unnecessary in product code.

The use case here is folks instrumenting per transaction stores where we don't need metrics at the per transaction level (cardinality too high and we have higher level metrics for that), but wanting tracing for analyzing at the request level

Reply:
> I'd gone back and forth on adding something like this initially -- it probably doesn't make sense to write a tracing-java-specific processor, however that would avoid the InvocationContext overhead incurred by tritium (that said, it would likely be best in the long term to evolve the tritium handler api to avoid that requirement, and centralize this sort of thing in one project)

Yeah, I debated adding something in tracing-java, but this felt lightweight enough to support `tracing` in addition to `instrument` (that adds metrics + tracing).